common/SmokeTests/SmokeTest/Update-Dependencies.ps1

```diff
@@ -31,63 +35,79 @@ function Log-Warning($message) {
     }
 }
 
-Register-PackageSource `
-    -Name $NIGHTLY_FEED_NAME `
-    -Location $NIGHTLY_FEED_URL `
-    -ProviderName Nuget `
-    -ErrorAction SilentlyContinue `
+function GetAllPackages {
+    $packages = Find-Package -Source $PACKAGE_FEED_URL -AllVersion -AllowPrereleaseVersions
+    if ($Daily) {
+        return $packages | Where-Object { $_.Version.Contains("alpha") }
+    }
+    return $packages
+}
 
-# List all packages from the source specified by $FeedName. Packages are sorted
-# ascending by version according to semver rules (e.g. 4.0.0-preview.1 comes
-# before 4.0.0) not lexicographically.
-# Packages cannot be filtered at this stage because the sleet feed to which they
-# are published does not support filtering by name.
-$allPackages = Find-Package -Source $NIGHTLY_FEED_NAME -AllVersion -AllowPrereleaseVersions
+function GetLatestPackage([array]$packageList, [string]$packageName) {
+    # This assumes that the versions coming back from Find-Package are
+    # sorted ascending. It excludes any version that is in the corresponding
+    # $PACKAGE_EXCLUSIONS entry
+    $versions = ($packageList `
+        | Where-Object { $_.Name -eq $packageName }
+        | Where-Object { `
```

Comment:
Tip: you can use `where-object $package_exclusisions.Keys -notcontains $packageName`. Might look a little cleaner.